src/parser.c

```diff
@@ -691,6 +716,169 @@ enum parse_rv parse_ExportTransaction(struct ExportTransactionState *const state
+          memset(&output_prompt, 0, sizeof(output_prompt));
+          if (!(meta->last_output_amount > 0)) REJECT("Assertion failed: last_output_amount > 0");
+          output_prompt.amount = meta->last_output_amount;
+          output_prompt.network_id = meta->network_id;
+          // addressState explicitly is not part of the ephemeral sub-states of
+          // EVMOutput_state so that this value is still available here.
+          memcpy(&output_prompt.address, &state->addressState.val,
+              sizeof(output_prompt.address));
+          
+          if(ADD_PROMPT(
+                "From X chain",
+                &output_prompt, sizeof(output_prompt),
+                output_prompt_to_string
+                ))
+              break;
+      }
+      case 4:
+        sub_rv=PARSE_RV_DONE;
+    }
+    return sub_rv;
+}
+
+IMPL_ARRAY(EVMOutput);
+
+void init_EVMInput(struct EVMInput_state *const state) {
+  state->state = 0;
+  INIT_SUBPARSER(addressState, Address);
+}
+
+enum parse_rv parse_EVMInput(struct EVMInput_state *const state, parser_meta_state_t *const meta) {
+    enum parse_rv sub_rv = PARSE_RV_INVALID;
+    switch (state->state) {
+      case 0: { // Address
+          CALL_SUBPARSER(addressState, Address);
+          state->state++;
+          PRINTF("ADDRESS: %.*h\n", 20, state->addressState.buf);
+          INIT_SUBPARSER(uint64State, uint64_t);
+      }
+      case 1: { // Amount
+          CALL_SUBPARSER(uint64State, uint64_t);
+          PRINTF("AMOUNT: %.*h\n", 8, &state->uint64State.val);
+            if (__builtin_uaddll_overflow(state->uint64State.val, meta->sum_of_inputs, &meta->sum_of_inputs)) THROW_(EXC_MEMORY_ERROR, "Sum of inputs overflowed");
+          state->state++;
+          INIT_SUBPARSER(id32State, Id32);
+      }
+      case 2: { // AssetID
+          CALL_SUBPARSER(id32State, Id32);
+          PRINTF("ASSET: %.*h\n", 32, state->id32State.buf);
+          INIT_SUBPARSER(uint64State, uint64_t);
+      }
+      case 3: { // nonce
+          CALL_SUBPARSER(uint64State, uint64_t);
+          PRINTF("NONCE: %.*h\n", 8, &state->uint64State.val);
+      }
+    }
+    return sub_rv;
+}
+
+IMPL_ARRAY(EVMInput);
+
+void init_CChainImportTransaction(struct CChainImportTransactionState *const state) {
+  state->state = 0;
+  INIT_SUBPARSER(uint32State, uint32_t);
+}
+
+enum parse_rv parse_CChainImportTransaction(struct CChainImportTransactionState *const state, parser_meta_state_t *const meta) {
+    enum parse_rv sub_rv = PARSE_RV_INVALID;
+      switch (state->state) {
+        case 0: // sourceChain 
+            CALL_SUBPARSER(id32State, Id32);
+            if(memcmp(network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->id32State.buf, sizeof(blockchain_id_t)))
+              REJECT("Invalid XChain ID");
+            state->state++;
+            INIT_SUBPARSER(inputsState, TransferableInputs);
+            PRINTF("Done with ChainID;\n");
+
+        case 1: { // PChain
```

Comment:
PChain ?

Reply:
Inaccurate, holdover from the X to P chain atomic swap I think.